### PR TITLE
Avoid traversing constants with Py_VISIT()

### PR DESCRIPTION
**Pull request by scoder**
Based off https://github.com/cython/cython/pull/6276 since it needs the same fix.

### CODE REVIEW

File: Cython/Utility/ModuleSetupCode.c [hunk 1]
```diff
@@ -1036,6 +1036,9 @@ static CYTHON_INLINE PyObject * __Pyx_PyDict_GetItemStrWithError(PyObject *dict,
   #define __Pyx_SET_REFCNT(obj, refcnt) Py_REFCNT(obj) = (refcnt)
   #define __Pyx_SET_SIZE(obj, size) Py_SIZE(obj) = (size)
 #endif
+// No-op macro for calling Py_VISIT() on known constants that can never participate in reference cycles.
+// Users can define it to "Py_VISIT(obj)" to help in debugging reference issues.
+#define __Pyx_VISIT_CONST(obj)
```
Comment (da-woods) [inline Cython/Utility/ModuleSetupCode.c:1041]: If you want users to be able to define it, would it make sense to do `ifndef`?

**Review by da-woods — COMMENTED** [review]
... otherwise looks good

**Comment by scoder** [comment]
>If you want users to be able to define it, would it make sense to do `ifndef`?

Ah, yes, right, thanks. That was the intention and I forgot to actually do it. :)

**Comment by scoder** [comment]
Added macro option, rebased on https://github.com/cython/cython/pull/6276

**Pull request merged**